nuxeo-features/nuxeo-automation/nuxeo-automation-server/src/main/java/org/nuxeo/ecm/automation/server/jaxrs/ResponseHelper.java

```diff
@@ -112,6 +113,9 @@ public static Object getResponse(Object result, HttpServletRequest request, int
             CoreSession session = SessionFactory.getSession(request);
             return Response.status(httpStatus).entity(((DocumentRefList) result).stream().map(session::getDocument)
                     .collect(Collectors.toCollection(DocumentModelListImpl::new))).build();
+        } else if (result instanceof List<?> && !((List) result).isEmpty()
+                && ((List) result).get(0) instanceof NuxeoPrincipal) {
+            return result;
```

Comment:
Could you return `httpStatus like in others condition ?
```suggestion
            return Response.status(httpStatus).entity(result).build();
```